internal/dag/gatewayapi_processor.go

```diff
@@ -1256,6 +1339,206 @@ func (p *GatewayAPIProcessor) computeHTTPRoute(route *gatewayapi_v1beta1.HTTPRou
 	return programmed, hosts
 }
 
+func (p *GatewayAPIProcessor) computeGRPCRoute(route *gatewayapi_v1alpha2.GRPCRoute, routeAccessor *status.RouteParentStatusUpdate, listener *listenerInfo) (bool, sets.Set[string]) {
+	hosts, errs := p.computeHosts(route.Spec.Hostnames, gatewayapi.HostnameDeref(listener.listener.Hostname))
+	for _, err := range errs {
+		// The Gateway API spec does not indicate what to do if syntactically
+		// invalid hostnames make it through, we're using our best judgment here.
+		// Theoretically these should be prevented by the combination of kubebuilder
+		// and admission webhook validations.
+		routeAccessor.AddCondition(gatewayapi_v1beta1.RouteConditionResolvedRefs, metav1.ConditionFalse, status.ReasonDegraded, err.Error())
+	}
+
+	// If there were no intersections between the listener hostname and the
+	// route hostnames, the route is not programmed for this listener.
+	if len(hosts) == 0 {
+		return false, nil
+	}
+
+	var programmed bool
+	for ruleIndex, rule := range route.Spec.Rules {
+		// Get match conditions for the rule.
+		var matchconditions []*matchConditions
+		for _, match := range rule.Matches {
+			// Convert method match to path match
+			pathMatch, ok := gatewayGRPCMethodMatchCondition(match.Method, routeAccessor)
+			if !ok {
+				continue
+			}
+
+			headerMatches, err := gatewayGRPCHeaderMatchConditions(match.Headers)
+			if err != nil {
+				routeAccessor.AddCondition(status.ConditionNotImplemented, metav1.ConditionTrue, status.ReasonHeaderMatchType, err.Error())
```

Comment:
Let's use `gatewayapi_v1beta1.RouteConditionAccepted` and `gatewayapi_v1beta1.RouteReasonUnsupportedValue` here as well.